### PR TITLE
[REF] web: remove legacy part of domjs

### PR DESCRIPTION
**Pull request by sauc-odoo**
Task-3586757

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/utils/ui.js [hunk 1]
```diff
@@ -227,3 +227,349 @@ export function addLoadingEffect(btnEl) {
...
+ * @param {jQuery} $from - the jQuery element(s) from which to search
+ * @param {string} selector - the CSS selector to match
+ * @param {boolean} [addBack=false] - whether or not the $from element
+ *                                  should be considered in the results
+ * @returns {jQuery}
+ */
+
+// TODO: Need to convert cssFind from jquery to vanilla js
+
+export function cssFind($from, selector, addBack) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/ui.js:251]: I would leave this function in legacy/dom.js as it uses jquery.

File: addons/web/static/src/core/utils/ui.js [hunk 2]
```diff
@@ -227,3 +227,349 @@ export function addLoadingEffect(btnEl) {
...
+ *        Note: automatically completed with "btn btn-X"
+ *        (@see options.size for the value of X)
+ * @param {string} [options.size] - @see options.attrs.class
+ * @param {string} [options.icon]
+ *        The specific fa icon class (for example "fa-home") or an URL for
+ *        an image to use as icon.
+ * @param {string} [options.text] - the button's text
+ * @returns {HTMLElement}
+ */
+export function renderButton(options) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/ui.js:288]: Do we still need this function? Maybe we could have a <Button> component instead

File: addons/web/static/src/core/utils/ui.js [hunk 3]
```diff
@@ -227,3 +227,349 @@ export function addLoadingEffect(btnEl) {
...
+}
+
+/**
+ * Calculates the offset position of the element relative to the document.
+ *
+ * @param {Object} options - Options for calculating the offset
+ * @returns {Array|Object} - Offset position(s) of the element
+ */
+function offset(options) {
+    var elements = this.length ? this : [this];
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/ui.js:349]: no more var, use const everywhere, and let when you can't

Comment (aab-odoo) [reply addons/web/static/src/core/utils/ui.js:349]: also, the API is really strange. I assume the function should take the element in argument, not treating `this` as the element

File: addons/web/static/src/core/utils/ui.js [hunk 4]
```diff
@@ -227,3 +227,349 @@ export function addLoadingEffect(btnEl) {
...
+ * @param {number} [options.extraOffset=0]
+ *      extra offset to add on top of the automatic one (the automatic one
+ *      being computed based on fixed header sizes)
+ * @param {number} [options.forcedOffset]
+ *      offset used instead of the automatic one (extraOffset will be
+ *      ignored too)
+ * @param {HTMLElement} [options.scrollable] the element to scroll
+ * @return {Promise}
+ */
+export function scrollTo(el, options = {}) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/utils/ui.js:461]: we have a utils/scrolling.js file (which already contains such a function)

Comment (sauc-odoo) [reply addons/web/static/src/core/utils/ui.js:417]: This scrollTo function is different from the one present in utils/scrolling.js file as it returns a promise that resolves when the scroll animation completes.

File: addons/web/static/src/core/utils/ui.js (hunk 3 above)
Comment (sauc-odoo) [reply addons/web/static/src/core/utils/ui.js]: Changes have been made, Please check it.
Thanks!!

File: addons/web/static/src/core/utils/ui.js (hunk 1 above)
Comment (sauc-odoo) [reply addons/web/static/src/core/utils/ui.js]: Hello @aab-odoo ,
Changes have been made, Please check it.
Thanks!!

File: addons/web/static/src/legacy/js/core/dialog.js [hunk 5]
```diff
@@ -293,7 +293,7 @@ var Dialog = Widget.extend({
         var self = this;
         $target.empty();
         buttons.forEach((buttonData) => {
-            var $button = dom.renderButton({
+            var button = renderButton({
```
Comment (mcm-odoo) [inline addons/web/static/src/legacy/js/core/dialog.js:296]: No `var` anymore, can be a `const` here.

File: addons/web_editor/static/src/js/editor/snippets.editor.js [hunk 6]
```diff
@@ -3707,7 +3707,8 @@ class SnippetsMenu extends Component {
         if (modalEl && !$(modalEl).hasScrollableContent()) {
             return;
         }
-        return dom.scrollTo($el[0], {extraOffset: 50, $scrollable: $scrollable});
+        let scrollable = $scrollable?.get(0);
```
Comment (mcm-odoo) [inline addons/web_editor/static/src/js/editor/snippets.editor.js:3710]:
```suggestion
        const scrollable = $scrollable?.get(0);
```

File: addons/web_editor/static/src/js/wysiwyg/widgets/image_crop.js [hunk 7]
```diff
@@ -252,15 +252,16 @@ export class ImageCrop extends Component {
         // HTML fields, the element to scroll is different from the document's
         // scrolling element).
         const $scrollable = closestScrollable(this.media);
+        let scrollable = $scrollable?.get(0);
```
Comment (mcm-odoo) [inline addons/web_editor/static/src/js/wysiwyg/widgets/image_crop.js:255]:
```suggestion
        const scrollable = $scrollable?.get(0);
```

**Comment by sauc-odoo** [comment]
>Hello, the `scrollTo` from `utils/scrolling.js` could return a promise too. We can listen to the event scrollend on the scrollable element and resolve a promise when triggered. This event would be listened just once.

Hello @mcm-odoo ,
I've pushed the suggested changes but some builds are failing in runbot - https://runbot.odoo.com/runbot/batch/1523544/build/64932118

I think it's happening because the `scrollTo` in `utils/scrolling.js` does not include animation behavior which was present in legacy dom.js `scrollTo` function -
https://github.com/odoo/odoo/blob/master/addons/web/static/src/legacy/js/core/dom.js#L222
and also in my implementation of vanilla js converted `scrollTo` function - 
https://github.com/odoo/odoo/commit/e8cab32a34c3fb2be7abbbc7f70a4d8b3cefe760#diff-e6555fb71d98ecd1d802b43ffa64e0580ac8012c0f2b5b3bf86a1a326914a965R459-R480

Kindly give your suggestions on this. Thanks!!

**Comment by sauc-odoo** [comment]
Hello @mcm-odoo 
All the discussed changes have been made. Please check them.
Thanks!!

**Comment by mcm-odoo** [comment]
@robodoo r+

**Comment by mcm-odoo** [comment]
@robodoo r+

**Pull request closed without merging**